script/main.js

```diff
@@ -0,0 +1,157 @@
+"use strict";
+
+class StockSearch {
+  constructor(showResultList) {
+    this.baseUrL = baseUrL;
+    this.getInput = document.getElementById("input");
+    this.getSearchBtn = document.getElementById("search");
+    this.showResultList = showResultList;
+    this.loadingSymbol = document.getElementById("loading");
+    this.errorsContainer = document.getElementById("errors");
+
+    this.getSearchBtn.addEventListener("click", () => this.getStockPrices());
+  }
+
+  loaderIndicator() {
+    this.loadingSymbol.classList.toggle("visually-hidden");
+  }
+
+  async getStockPrices() {
+    this.loaderIndicator();
+    try {
+      let response = await fetch(
+        `${this.baseUrL}/api/v3/search?query=${this.getInput.value}&limit=10&exchange=NASDAQ`
+      );
+      let data = await response.json();
+      this.makeResultList(data);
+    } catch (error) {
+      this.errorsContainer.innerHTML = `Error: ${error.message}`;
+    } finally {
+      this.loaderIndicator();
+    }
+  }
+
+  highlightMatches(text, query) {
+    const regex = new RegExp(query, "gi");
+    return text.replace(regex, match => `<mark id="mark">${match}</mark>`);
+  }
+
+  makeResultList(data) {
```

Comment:
this function should be inside another class that handles the rendering of the list of results.